Src/FluentAssertions/Equivalency/DataRelationEquivalencyStep.cs

```diff
@@ -0,0 +1,202 @@
+                        .ForCondition(subject.RelationName == expectation.RelationName)
+                        .FailWith($"Expected RelationName to be '{expectation.RelationName}', but found '{subject.RelationName}'");
+                }
+
+                if (selectedMembers.ContainsKey(nameof(expectation.Nested)))
+                {
+                    AssertionScope.Current
+                        .ForCondition(subject.Nested == expectation.Nested)
+                        .FailWith($"Expected Nested to be '{expectation.Nested}', but found '{subject.Nested}'");
+                }
+
+                // Special case: Compare name only
+                if (selectedMembers.ContainsKey(nameof(expectation.DataSet)))
+                {
+                    AssertionScope.Current
+                        .ForCondition(subject.DataSet?.DataSetName == expectation.DataSet?.DataSetName)
+                        .FailWith($"Expected containing DataSet to be {expectation.DataSet?.DataSetName ?? "<null>"}, but found {subject.DataSet?.DataSetName ?? "<null>"}");
+                }
+
+                // Collections
+                if (selectedMembers.TryGetValue(nameof(expectation.ExtendedProperties), out var expectationMember))
+                {
+                    var matchingMember = FindMatchFor(expectationMember, context, config);
+
+                    if (matchingMember != null)
+                    {
+                        IEquivalencyValidationContext nestedContext =
+                            context.CreateForNestedMember(expectationMember, matchingMember);
+
+                        if (nestedContext != null)
+                        {
+                            parent.AssertEqualityUsing(nestedContext);
+                        }
+                    }
+                }
+
+                // Relation constraints
+                CompareConstraints(
+                    parent, context, config, subject, expectation, selectedMembers,
+                    "Child",
+                    selectedMembers.ContainsKey(nameof(DataRelation.ChildTable)),
+                    selectedMembers.ContainsKey(nameof(DataRelation.ChildColumns)),
+                    selectedMembers.ContainsKey(nameof(DataRelation.ChildKeyConstraint)),
+                    r => r.ChildColumns,
+                    r => r.ChildTable);
+
+                CompareConstraints(
+                    parent, context, config, subject, expectation, selectedMembers,
+                    "Parent",
+                    selectedMembers.ContainsKey(nameof(DataRelation.ParentTable)),
+                    selectedMembers.ContainsKey(nameof(DataRelation.ParentColumns)),
+                    selectedMembers.ContainsKey(nameof(DataRelation.ParentKeyConstraint)),
+                    r => r.ParentColumns,
+                    r => r.ParentTable);
+            }
+
+            return true;
+        }
+
+        private static void CompareConstraints(
```

Comment:
🤔 I think this method can become a lot more readable if you break it down into three. One for comparing columns, one for comparing tables and one for comparing key constraints.

Reply:
Done :-)